### PR TITLE
Adding mdc support for LoggingReceive #21361

### PR DESCRIPTION
**Pull request by ortibazar**

### CODE REVIEW

File: akka-actor-tests/src/test/scala/akka/event/LoggingReceiveSpec.scala [hunk 1]
```diff
@@ -118,6 +118,23 @@ class LoggingReceiveSpec extends WordSpec with BeforeAndAfterAll {
...
+        system.eventStream.subscribe(testActor, classOf[Logging.Debug])
+        val myMDC = Map("hello" → "mdc")
+        val a = system.actorOf(Props(new Actor with DiagnosticActorLogging {
+          override def mdc(currentMessage: Any) = myMDC
+          def receive = LoggingReceive {
+            case "hello" ⇒
+          }
+        }))
+        a ! "hello"
+        expectMsgPF(1 second, "Logging.Debug2") {
```
Comment (patriknw) [inline akka-actor-tests/src/test/scala/akka/event/LoggingReceiveSpec.scala]: too short timeout, see https://github.com/akka/akka/blob/master/CONTRIBUTING.md#preferred-ways-to-use-timeouts-in-tests

**Comment by patriknw** [comment]
LGTM, aside from a test timeout
Excellent contribution @ortigali !

**Comment by patriknw** [comment]
OK TO TEST

**Comment by johanandren** [comment]
LGTM

**Comment by johanandren** [comment]
Refs #21361

**Pull request merged**